heron/executor/src/python/heron_executor.py

```diff
@@ -539,40 +564,32 @@ def get_command_changes(self, current_commands, updated_commands):
     return commands_to_kill, commands_to_keep, commands_to_start
 
   def launch(self):
-    current_commands = dict(map((lambda x: (x[1], x[2])), self.processes_to_monitor.values()))
-    updated_commands = self.get_commands_to_run()
-
-    # get the commands to kill, keep and start
-    commands_to_kill, commands_to_keep, commands_to_start = \
-        self.get_command_changes(current_commands, updated_commands)
-
-    Log.info("current commands: %s" % sorted(current_commands.keys()))
-    Log.info("new commands    : %s" % sorted(updated_commands.keys()))
-    Log.info("commands_to_kill: %s" % sorted(commands_to_kill.keys()))
-    Log.info("commands_to_keep: %s" % sorted(commands_to_keep.keys()))
-    Log.info("commands_to_start: %s" % sorted(commands_to_start.keys()))
-
-    self._kill_processes(commands_to_kill)
-    self._start_processes(commands_to_start)
-    Log.info("Launch complete - processes killed=%s kept=%s started=%s monitored=%s" %
-             (len(commands_to_kill), len(commands_to_keep),
-              len(commands_to_start), len(self.processes_to_monitor)))
-
-  def prepare_launch(self):
-    create_folders = 'mkdir -p %s' % self.log_dir
-    chmod_binaries = \
-        'chmod a+rx . && chmod a+x %s && chmod +x %s && chmod +x %s && chmod +x %s' \
-        % (self.log_dir, self.tmaster_binary, self.stmgr_binary, self.heron_shell_binary)
+    with self.process_lock:
+      current_commands = dict(map((lambda x: (x[1], x[2])), self.processes_to_monitor.values()))
+      updated_commands = self.get_commands_to_run()
 
-    commands = [create_folders, chmod_binaries]
+      # get the commands to kill, keep and start
+      commands_to_kill, commands_to_keep, commands_to_start = \
+          self.get_command_changes(current_commands, updated_commands)
+
+      Log.info("current commands: %s" % sorted(current_commands.keys()))
+      Log.info("new commands    : %s" % sorted(updated_commands.keys()))
+      Log.info("commands_to_kill: %s" % sorted(commands_to_kill.keys()))
+      Log.info("commands_to_keep: %s" % sorted(commands_to_keep.keys()))
+      Log.info("commands_to_start: %s" % sorted(commands_to_start.keys()))
```

Comment:
Yeah, I think this are ok as info. They happen rarely and we typically want to know what's going on with this process restarting part.